pkgs/by-name/fu/fuse-t/package.nix

```diff
@@ -0,0 +1,185 @@
+    buildInputs = [
+      libiconv.out
+      DiskArbitration
+    ];
+
+    propagatedBuildInputs = [ DiskArbitration ];
+
+    # Removes a lot of warning spam
+    # `warning: 'pthread_setugid_np' is deprecated: Use of per-thread security contexts is error-prone and discouraged.`
+    env.NIX_CFLAGS_COMPILE = "-Wno-deprecated-declarations";
+
+    postBuild = ''
+      mkdir -p lib/pkgconfig
+      tapi stubify --filetype=tbd-v2 lib/libfuse-t.dylib -o lib/libfuse-t.tbd
+      substituteAll ${./fuse-t.pc.in} lib/pkgconfig/fuse-t.pc
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      frameworkDir=$out/Library/Frameworks/fuse_t.framework
+      frameworkADir=$frameworkDir/Versions/A
+      mkdir -p $out/lib/pkgconfig $out/include/fuse $frameworkADir/Headers $frameworkADir/PrivateHeaders
+      cp lib/libfuse-t.a lib/libfuse-t.dylib lib/libfuse-t.tbd $out/lib
+      cp lib/pkgconfig/fuse-t.pc $out/lib/pkgconfig
+      cd ..
+      cp include/fuse* $out/include/fuse
+      cp include/fuse* $frameworkADir/Headers
+      cp include/fuse* $frameworkADir/PrivateHeaders
+      ln -s A $out/Library/Frameworks/fuse_t.framework/Versions/Current
+      ln -s Versions/Current/Headers $frameworkDir
+      ln -s Versions/Current/PrivateHeaders $frameworkDir
+      cp include/config.h include/cuse_*.h lib/fuse_*.h $frameworkADir/PrivateHeaders
+      cp include/cuse_*.h $frameworkADir/PrivateHeaders
+      cp lib/fuse_*.h $frameworkADir/PrivateHeaders
+      runHook postInstall
+    '';
+
+    doCheck = false;
+
+    meta = with lib; {
+      description = "Library that allows filesystems to be implemented in user space";
+      longDescription = ''
+        FUSE (Filesystem in Userspace) is an interface for userspace programs to
+        export a filesystem to the Linux kernel. The FUSE project consists of two
+        components: The fuse kernel module (maintained in the regular kernel
+        repositories) and the libfuse userspace library (this package). libfuse
+        provides the reference implementation for communicating with the FUSE
+        kernel module.
+      '';
+      homepage = "https://github.com/macos-fuse-t/libfuse";
+      platforms = platforms.darwin;
+      license = licenses.lgpl21Only;
+    };
+  };
+in
+stubs.overrideAttrs {
+  pname = "fuse-t";
+
+  postInstall = ''
+    cp -RHnv ${binaryDistribution}/. $out
+    rm $out/lib/libfuse-t-${version}.a $out/lib/libfuse-t-${version}.dylib
+  '';
+
+  passthru = {
+    inherit binaryDistribution stubs;
+  };
+
+  meta = {
+    description = "FUSE-T is a kext-less implementation of FUSE for macOS that uses NFS v4 local server instead of a kernel extension.";
+    homepage = "https://www.fuse-t.org/";
+    license = [
+      # libfuse/stubs
+      lib.licenses.lgpl21Only
+
+      # NFS server
+      # free for non-commercial use, see: https://github.com/macos-fuse-t/fuse-t/blob/main/License.txt
+      lib.licenses.unfreeRedistributable
```

Comment:
I'm a bit concerned about including the unfree NFS server. I believe Hydra doesn't build unfreeRedistributable packages. Wouldn't this cause more popular packages to be dropped from the Darwin binary cache as more packages switch to FUSE-T?

https://github.com/NixOS/nixpkgs/issues/83884

Reply:
AIUI packages will only link to the stubs, which are FOSS (as they have to be, because FUSE is LGPL).

(Will try to find the time to review the rest of this soon.)